src/store/index.ts

```diff
@@ -4,10 +4,21 @@
 
 import { createStore } from "vuex"
 
+const DEFAULT_NUMERIC_FORMAT_LANGUAGE = "fr-FR"
+function mkNumericFormat(language: string) {
+  return new Intl.NumberFormat(language, {
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+  })
+}
+
+
 export default createStore({
   state: {
     showCredit: false,
     isModalOpen: false,
+    numericFormatLanguage: DEFAULT_NUMERIC_FORMAT_LANGUAGE,
+    numericFormat: mkNumericFormat(DEFAULT_NUMERIC_FORMAT_LANGUAGE),
```

Comment:
I was not super sure I understood your comment. In the highlighted lines, we store a formatter object (the result of the `mkNumericFormat` method call). So we don't have a method in vuex state, but a method call result. I figured that you were probably refering to the fact that this object is still quite complex (and holds the `.format(..)` method) ? And I agree that it would be preferrable to have simple values in `state`. I didn't see obvious way at first to store elegantly the instance to avoid getting it to be re-instantiated and in the meantime being accessible to all components.

But I found another solution that will use `getters` and the caching system provided naturally by `vuex` stores : It should actually not run the getter twice (as long as `numericFormatLanguage` didn't change, which is exactly what we want), and as such, should not instanciate twice the formatter.

While implementing this idea, I found out that `vuex` doesn't cache `getters` as it should (cf the warning note in https://vuex.vuejs.org/guide/getters.html#getters ). So I had to make a quick cache system that will be removable when we'll switch to pinia or when vuex will correct this bug. 

So I pushed the fixes related to your comment in separate commit at the end. Tell me what do you think of them ;-)

Reply:
Maybe I did not make myself clear you're right. I understood that what was stored was the `Intl.NumberFormat` and that it was built from a factory. That was what I was referring to when I wrote "I see why you want to do that".

What stroke me at first was the use of the store alltogether. I still would have done it without the store, maybe with a native observer, or maybe not reactive at all since we're not changing the numericFormat while displaying formatted values, but I see your solution handles all cases upfront. In my mind, this object still does not fall into the "put it in the store" category but this is just an expression of my own mental limitation. I never did smth like that, maybe now I would. And your final solution is really clever.

Thanks for keeping with me, and thx for the heads-up, and I was not aware of Pinia